cpp/src/Slice/Grammar.y

```diff
@@ -1674,41 +1689,33 @@ enumerator
 : ICE_IDENTIFIER
 {
     auto ident = dynamic_pointer_cast<StringTok>($1);
-    auto ens = make_shared<EnumeratorListTok>();
-    ContainerPtr cont = currentUnit->currentContainer();
-    EnumeratorPtr en = cont->createEnumerator(ident->v, nullopt);
-    if (en)
-    {
-        ens->v.push_front(en);
-    }
-    $$ = ens;
+    EnumPtr cont = dynamic_pointer_cast<Enum>(currentUnit->currentContainer());
+    $$ = cont->createEnumerator(ident->v, nullopt);
 }
 | ICE_IDENTIFIER '=' enumerator_initializer
 {
     auto ident = dynamic_pointer_cast<StringTok>($1);
-    auto ens = make_shared<EnumeratorListTok>();
-    ContainerPtr cont = currentUnit->currentContainer();
+    EnumPtr cont = dynamic_pointer_cast<Enum>(currentUnit->currentContainer());
     auto intVal = dynamic_pointer_cast<IntegerTok>($3);
     if (intVal)
     {
         if (intVal->v < 0 || intVal->v > std::numeric_limits<int32_t>::max())
         {
             currentUnit->error("value for enumerator `" + ident->v + "' is out of range");
         }
-        else
-        {
-            EnumeratorPtr en = cont->createEnumerator(ident->v, static_cast<int>(intVal->v));
-            ens->v.push_front(en);
-        }
+        $$ = cont->createEnumerator(ident->v, static_cast<int>(intVal->v));
+    }
+    else
+    {
+        $$ = cont->createEnumerator(ident->v, nullopt); // Dummy
```

Comment:
I don't understand the dummy comment.

Reply:
This case only occurs when the cast `auto intVal = dynamic_pointer_cast<IntegerTok>($3)` fails.
But this grammar rule is only triggered when there's an integer literal present...

So something must of gone wrong for the cast to fail. To be honest, I didn't look too closely at this specific case,
but it might just be impossible to end up in this case after thinking about it now.